[compliance] Package security-agent-policies compliance configs

xornivore opened the pull request:
### What does this PR do?

Packaging default compliance benchmarks from a security-agent-policies repo in agent packages,

This mimics pulling a version of integrations-core, there is however no install to do - just a file copy. 

### Motivation

Meaningful configuration out of the box for compliance.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Build agent images, confirm benchmarks are correctly packaged under `/etc/datadog-agent/compliance.d`

Commit:
[compliance] Package security-agent-policies compliance configs

Commit:
Package security agent compliance policy files in the DCA image

Commit:
Change repo to the official security-agent-policies

Commit:
Add release note

KSerrania reviewed (commented):
I think you'll want to declare joint ownership between Agent Platform and your team in `CODEOWNERS` for the new `omnibus/config/software/datadog-security-agent-policies.rb` file, same as what we do here: https://github.com/DataDog/datadog-agent/blob/c879895b994a6d1ddc3826397b7aec25a0790add/.github/CODEOWNERS#L76-L77

(otherwise you'll need a review from us every time you want to change something in this software definition).

ahmed-mez commented on tasks/cluster_agent.py:
```diff
@@ -40,6 +44,24 @@ def build(ctx, rebuild=False, build_include=None, build_exclude=None, race=False
         skip_assets,
     )
 
+    policies_version = "master"
```
should we make this configurable? i.e could it be useful to package a custom policies version to build a custom agent version for example?

xornivore replied:
Yes, definitely. Made it configurable in the task.

xornivore commented on tasks/cluster_agent.py:
```diff
@@ -40,6 +44,24 @@ def build(ctx, rebuild=False, build_include=None, build_exclude=None, race=False
 
+    policies_version = "master"
+
+    git_ref = ctx.run("git rev-parse --abbrev-ref HEAD", hide=True).stdout.strip()
+    release_version = ctx.run("git describe --always --contains {}".format(git_ref), hide=True).stdout.strip()
```
Oh, great, yes, that's much better, thank you! I believe it should also cover @ahmed-mez's comment about making it configurable - so the release-version is a param you can use that as well.

Commit:
Address review feedback/comments

- Add `policies_version` and `release_version` args to cluster-agent.build to make it configurable in CI and manual runs.
- Use config for `nightly-a7` by default and `CI_COMMIT_TAG` if set for `dca-*`.
- Update `CODEOWNERS` for `omnibus/config/software/datadog-security-agent-policies.rb`

prognant reviewed (approved):
LGTM
